test/functional/mongo_client_tests.js

```diff
@@ -415,7 +415,7 @@ exports["should correctly connect to mongodb using domain socket"] = {
   // The actual test we wish to run
   test: function(configuration, test) {
     var MongoClient = configuration.require.MongoClient;
-    MongoClient.connect('mongodb:///tmp/mongodb-27017.sock/test', function(err, client) {
+    MongoClient.connect('mongodb://%2Ftmp%2Fmongodb-27017.sock/test', function(err, client) {
```

Comment:
what's going on here? I don't think its correct that we require users to escape their own local domain socket paths, are you saying it doesn't pass the connection string tests to do so? We might need to file a spec ticket against that then